cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -158,121 +158,131 @@ class CommonCliffordGateMetaClass(value.ABCMetaImplementAnyOneOf):
         return cls._CZ
 
     @property
     def SWAP(cls) -> 'cirq.CliffordGate':
         if not hasattr(cls, '_SWAP'):
-            cls._SWAP = CliffordGate.from_clifford_tableau(_gate_tableau(2, common_gates.SWAP))
+            t = qis.CliffordTableau(num_qubits=2)
+            t.xs = [[0, 1], [1, 0], [0, 0], [0, 0]]
+            t.zs = [[0, 0], [0, 0], [0, 1], [1, 0]]
+            cls._SWAP = CliffordGate.from_clifford_tableau(t)
         return cls._SWAP
 
     @property
     def X_sqrt(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_X_sqrt'):
-            # Unfortunately, due the code style, the matrix should be viewed transposed.
-            # Note xs, zs, and rs are column vector.
-            # Transformation: X -> X, Z -> -Y
-            clifford_tableau = qis.CliffordTableau._from_json_dict_(
-                n=1, rs=[0, 1], xs=[[1], [1]], zs=[[0], [1]]
-            )
-            cls._X_sqrt = SingleQubitCliffordGate.from_clifford_tableau(clifford_tableau)
-        return cls._X_sqrt
+        return cls.all_single_qubit_cliffords[4]
 
     @property
     def X_nsqrt(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_X_nsqrt'):
-            # Transformation: X->X, Z->Y
-            clifford_tableau = qis.CliffordTableau._from_json_dict_(
-                n=1, rs=[0, 0], xs=[[1], [1]], zs=[[0], [1]]
-            )
-            cls._X_nsqrt = SingleQubitCliffordGate.from_clifford_tableau(clifford_tableau)
-        return cls._X_nsqrt
+        return cls.all_single_qubit_cliffords[7]
 
     @property
     def Y_sqrt(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_Y_sqrt'):
-            # Transformation: X -> -Z,  Z -> X
-            clifford_tableau = qis.CliffordTableau._from_json_dict_(
-                n=1, rs=[1, 0], xs=[[0], [1]], zs=[[1], [0]]
-            )
-            cls._Y_sqrt = SingleQubitCliffordGate.from_clifford_tableau(clifford_tableau)
-        return cls._Y_sqrt
+        return cls.all_single_qubit_cliffords[5]
 
     @property
     def Y_nsqrt(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_Y_nsqrt'):
-            # Transformation: X -> Z, Z -> -X
-            clifford_tableau = qis.CliffordTableau._from_json_dict_(
-                n=1, rs=[0, 1], xs=[[0], [1]], zs=[[1], [0]]
-            )
-            cls._Y_nsqrt = SingleQubitCliffordGate.from_clifford_tableau(clifford_tableau)
-        return cls._Y_nsqrt
+        return cls.all_single_qubit_cliffords[8]
 
     @property
     def Z_sqrt(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_Z_sqrt'):
-            # Transformation: X -> Y, Z -> Z
-            _clifford_tableau = qis.CliffordTableau._from_json_dict_(
-                n=1, rs=[0, 0], xs=[[1], [0]], zs=[[1], [1]]
-            )
-            cls._Z_sqrt = SingleQubitCliffordGate.from_clifford_tableau(_clifford_tableau)
-        return cls._Z_sqrt
+        return cls.all_single_qubit_cliffords[6]
```

Comment:
This is the same as `S` on line 234.  Not sure which one is a typo.

Can you perhaps add an independent test that class-property gates have correct values?
If you keep the current property implementation with indexed lookup, such test will fail when
all_single_qubit_cliffords goes out of order.

Reply:
The S gate and Z_sqrt are the same gate. Square the S gate to convince yourself of that.

The PR does include independent test (test_all_single_qubit_clifford_unitaries) that checks all the gates have correct values. It is already order-sensitive.